Feature flu square_path

joachimhgg opened the pull request:
- [x] (347) samples/robotbenchmark/maze_runner/protos/MazeBlock.proto
- [x] (348) samples/robotbenchmark/square_path/protos/SquarePathPioneer3dx.proto
- ##SUPERVISOR (349) samples/robotbenchmark/square_path/protos/SquarePathBenchmark.proto
- [x] (356) samples/robotbenchmark/wall_following/protos/WallFollowingPioneer3dx.proto
- [x] (357) samples/robotbenchmark/wall_following/protos/LinkedWalls.proto
- [x] (068) projects/samples/robotbenchmark/square_path/worlds/square_path.wbt
- [x] (074) projects/samples/robotbenchmark/wall_following/worlds/wall_following.wbt

lukicdarkoo commented:
The world (`projects/samples/robotbenchmark/square_path/worlds/square_path.wbt`) is not converted.

joachimhgg commented:
> The world (`projects/samples/robotbenchmark/square_path/worlds/square_path.wbt`) is not converted. 
It has been converted at the commit [8d271f7935573a09a879ba8299ec5f25f4089e00](https://github.com/cyberbotics/webots/pull/3651/commits/8d271f7935573a09a879ba8299ec5f25f4089e00)

lukicdarkoo reviewed (dismissed):
Some worlds in which the PROTOs are used are not converted. For example:
- `VisualTrackingBenchmark` in `projects/samples/robotbenchmark/visual_tracking/worlds/visual_tracking.wbt`
- `RobotProgrammingBenchmark` in `projects/samples/robotbenchmark/robot_programming/worlds/robot_programming.wbt`

joachimhgg commented:
> Some worlds in which the PROTOs are used are not converted. For example:
> 
>     * `VisualTrackingBenchmark` in `projects/samples/robotbenchmark/visual_tracking/worlds/visual_tracking.wbt`
> 
>     * `RobotProgrammingBenchmark` in `projects/samples/robotbenchmark/robot_programming/worlds/robot_programming.wbt`

These worlds are depending of other robots. I modified the PR description and I will create other PRs for these.

ad-daniel commented on projects/samples/robotbenchmark/square_path/protos/SquarePathPioneer3dx.proto:
```diff
@@ -4,7 +4,7 @@
 
 PROTO SquarePathPioneer3dx [
-  field SFVec3f    translation     0 0.0975 0                       # Is `Transform.translation`.
+  field SFVec3f    translation     0 0 0.0975                      # Is `Transform.translation`.
```
Changes to the PROTO header should also be updated in the corresponding documentation of the robot

ad-daniel commented on projects/samples/robotbenchmark/square_path/protos/SquarePathPioneer3dx.proto:
```diff
@@ -721,7 +721,7 @@ PROTO SquarePathPioneer3dx [
               }
               Transform {
-                rotation 0 0 1 1.5708
+                rotation -1 0 0 -1.570796
```
```suggestion
                rotation 1 0 0 1.570796
```

ad-daniel commented on projects/samples/robotbenchmark/square_path/protos/SquarePathPioneer3dx.proto:
```diff
@@ -4,7 +4,7 @@
 # The Pioneer 3-DX robot (Adept MobileRobots) is an all-purpose base, used for research and applications involving mapping, teleoperation, localization, monitoring, reconnaissance and other behaviors.
```
There appears to be a difference compared to the master branch, something seems to be in front of the distance sensors
![a](https://user-images.githubusercontent.com/44834743/135426815-f1508ca1-05f6-4b4d-b791-a0082a9065a3.png)
![b](https://user-images.githubusercontent.com/44834743/135426823-34c41753-f5b3-41c8-b890-45d7c027d0c0.png)

I don't know if it's just a visual difference or if it actually impacts the sensors. Any clue what that might be?

joachimhgg replied:
It was just a visual difference (problem with translations). It's fixed.

ad-daniel commented on projects/samples/robotbenchmark/square_path/worlds/square_path.wbt:
```diff
@@ -8,20 +8,19 @@ WorldInfo {
-  translation 0 0.0943896 0
-  rotation 0 1 0 4.7123889
+  translation 0 0 0.09439
+  rotation 0 0 1 0
```
```suggestion
```

ad-daniel commented on projects/samples/robotbenchmark/square_path/worlds/square_path.wbt:
```diff
@@ -49,4 +48,4 @@ RectangleArena {
   }
   wallHeight 0.5
-}
+}
```
```suggestion
}

```

ad-daniel commented on projects/samples/robotbenchmark/wall_following/protos/WallFollowingPioneer3dx.proto:
```diff
@@ -722,7 +722,7 @@ PROTO WallFollowingPioneer3dx [
               }
               Transform {
-                rotation 0 0 1 1.5708
+                rotation -1 0 0 -1.570796
```
```suggestion
                rotation 1 0 0 1.570796
```

ad-daniel commented on projects/samples/robotbenchmark/wall_following/protos/WallFollowingPioneer3dx.proto:
```diff
@@ -712,7 +712,7 @@ PROTO WallFollowingPioneer3dx [
             children [
               Transform {
-                rotation 0 0 1 1.5708
+                rotation -1 0 0 -1.570796
```
```suggestion
                rotation 1 0 0 1.570796
```

ad-daniel replied:
ditto.

ad-daniel commented on projects/samples/robotbenchmark/wall_following/protos/WallFollowingPioneer3dx.proto:
```diff
@@ -4,8 +4,8 @@
-  field SFVec3f    translation     0 0.0975 0                       # Is `Transform.translation`.
-  field SFRotation rotation        0 0 1 0                          # Is `Transform.rotation`.
+  field SFVec3f    translation     0 0 0.0975                       # Is `Transform.translation`.
+  field SFRotation rotation        0 0 1 0                         # Is `Transform.rotation`.
```
```suggestion
  field SFRotation rotation        0 0 1 0                          # Is `Transform.rotation`.
```

joachimhgg commented:
> Looks good, just missing docs and some minor doubts.
> 
> The `wall_following.wbt` world cannot be converted in this PR?

It seems the doc were already converted. I fixed the issue and converted wall_following.wbt.

ad-daniel reviewed (approved):
Thank you, the broken tests are unrelated to this PR
